cmd/agent/subcommands/streamlogs/command.go

```diff
@@ -74,6 +100,19 @@ func streamLogs(log log.Component, config config.Component, cliParams *cliParams
 	urlstr := fmt.Sprintf("https://%v:%v/agent/stream-logs", ipcAddress, config.GetInt("cmd_port"))
 	return streamRequest(urlstr, body, func(chunk []byte) {
 		fmt.Print(string(chunk))
+
+		if cliParams.FilePath != "" {
+			f, bufWriter, err := writeToFile(cliParams.FilePath, string(chunk))
+			if err != nil {
+				fmt.Printf("Error writing stream-logs to file %s: %v", cliParams.FilePath, err)
+			}
+			defer f.Close()
```

Comment:
I think this is almost what we want, but not quite. 

`defer f.Close()` needs to be outside `streamRequest` since this function is called once for each chunk of logs. 

The ideal workflow would be:

1. Open file for writing
2. as slogs stream in `streamRequest` continuously write to the file
3. when the stream is done/cancelled close the file handle.